### ISSUE TITLE
Project Run Issue in Meteor?

### ISSUE THREAD

**Issue by kasireddynv** [sdarnell/meteor-win#33]
I start to run Meteor project in command prompt as shown below

```
C:\Users\dsk>e:

E:\>cd abc/red

E:\abc/red>meteor //Here I got the below error then click any button the error popup and close the error popup and doesn't runmeteor project

E:\abc/red>  //here some times get this (or) some time gets another

     (or)

 Exited with code: -1073741819
=> Your application is crashing. Waiting for file change.
```

![enter image description here](http://i.stack.imgur.com/yaNl9.png)

So please see the error and suggest me what to do?

**Comment by sdarnell** [comment]
Which operating system are you using and which version of Meteor?
From the command line, type "meteor --version" to get the version of meteor running the app.

This error did occur a long time ago - but was fixed in a change to fibers, so I'm a little surprised it has resurfaced.

Another possibility is that you have an old version installed and that is overriding a newer one.

Also, I notice you have a mix of forward and backward slashes in "E:\abc/red>" is this normal, or a typo? (I doubt it is related to the issue in question)

**Comment by kasireddynv** [comment]
I am using the "Windows 7 Operator System" and Doesn't work any command of Meteor.and I am using the Meteor version hope to think is above 6.

**Comment by sdarnell** [comment]
Check your PATH for anything that looks like meteor. Type 'where meteor'.
Old versions of meteor used to be in %ProgramFiles% but newer versions use %LOCALAPPDATA%.

Please check whether you have meteor in your add/remove programs list. If it is, I think you want to uninstall it as it will be an old version.

**Comment by kasireddynv** [comment]
```
    C:\Users\dsk>e:
    E:\>where meteor
    C:\Users\dsk\AppData\Local\.meteor\meteor.bat
    E:\>
```

Yeah, I find out as you told above Please check once above one?.

**Comment by sdarnell** [comment]
Right, I think the best solution is to remove the old meteor and start afresh.
Please delete the entire directory: C:\Users\dsk\AppData\Local.meteor
Then re-install meteor using LaunchMeteor.exe from win.meteor.com
which is actually: http://win-install.meteor.com/LaunchMeteor.exe

Then try again :)

If this doesn't fix the issue, then I think there must be a clash with some other installation of node on your machine. But hopefully this will do the trick.

**Comment by kasireddynv** [comment]
Okay , Thank You.

**Comment by kasireddynv** [comment]
Yeah, I have deleted previous one then download as new from win.meteor.com and try to install but again got the "same problem". What i am get at the installation time as shown below.

```
Downloading initial Meteor files...
100    
###########################################################################
Download complete (33.7 MB)
Extracting files to C:\Users\dsk\AppData\Local\.meteor
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
...........
    Extracted 6586 files (110.1 MB)
 Files extracted successfully
```

And get the same error popup as mentioned above.
Can you please suggest me what to do for this problem?

**Comment by sdarnell** [comment]
Please could you search your whole disk(s) for a file called "fibers.node". There should only be one.
For example type "dir c:\fibers.node /s/b"  (without the quotes) will search the C: drive and print any files.

Please could you also send the contents of your PATH environment variable, and anything starting NODE. To do this, just type "set PATH" and then "set NODE".

**Comment by kasireddynv** [comment]
Yeah , fibers.node is there only one.And the contents of your PATH environment variable as shown below

```
C:\Users\dsk\AppData\Local\.meteor\tools\47782522f9\lib\node_modules\fibers\bin
\win32-ia32-v8-3.14>set PATH
Path=C:\Program Files\nodejs\;C:\Users\dsk\AppData\Local\.me
teor;C:\Users\dsk\AppData\Roaming\npm
PATHEXT=.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC

C:\Users\dsk\AppData\Local\.meteor\tools\47782522f9\lib\node_modules\fibers\bin
\win32-ia32-v8-3.14>set NODE
Environment variable NODE not defined
```

**Comment by sdarnell** [comment]
Hmm, please could you try removing "C:\Program Files\nodejs\" from your PATH and try again.
It might also be worth removing npm too (as a test)

set PATH=C:\Users\dsk\AppData\Local.meteor

**Comment by kasireddynv** [comment]
Yeah, I removed as you told above and add also yet not working suffer from same error.Can you please give any alternate solution for this?

**Comment by sdarnell** [comment]
Could you confirm exactly which version of win7 you are running.
Is it 32-bit or 64-bit?
Do you have the latest windows updates?

You can get much of this information by running systeminfo.exe, for example here's a selection of info from my laptop:

```
OS Name:                   Microsoft Windows 7 Home Premium
OS Version:                6.1.7601 Service Pack 1 Build 7601
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
...
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 42 Stepping 7 GenuineIntel ~1601 Mhz
...
Hotfix(s):                 240 Hotfix(s) Installed.
```

The other thing to investigate is exactly what executables are getting loaded during an invocation of meteor. For this I would suggest using tool ProcessMonitor from Microsoft/SysInternals: http://technet.microsoft.com/en-gb/sysinternals/bb896645.aspx

Start processmo.exe and enable just Process and thread activity.
![image](https://f.cloud.github.com/assets/3684334/2331178/6cd450a4-a445-11e3-9c73-c4214c678313.png)

Then start meteor from the command line, after it fails, stop capturing in procmon, and scroll down in procmon to find where you find node being invoked:
![image](https://f.cloud.github.com/assets/3684334/2331201/c720fc92-a445-11e3-8053-5d653d7dde8d.png)

It will load lots of DLLs before getting to fibers.node:
![image](https://f.cloud.github.com/assets/3684334/2331209/fd41ccac-a445-11e3-95d4-151ec051feee.png)

If that doesn't show anything odd, you can select the fibers.node line and then enable the other views, such as the 'file system activity' and registry to see if anything weird shows up. But to be fair, if you have not looked at this level it may not be obvious.

You may want to send me some of these details, and although github is great, it may be easier to email me directly. You can find my email on my github user page.

**Comment by mgeeforce** [comment]
I am running into the same issue. These guys seem to shed some light on it: https://github.com/laverdet/node-fibers/issues/122

**Comment by sdarnell** [comment]
Looks like the thread locals count just needs extending a bit, and having had to tinker with this code previously, I know what to do, and had to rebuilt fibers previously.
The interesting thing is that on some XP systems it does work fine, but presumably other things installed seem to impact where the thread locals go on XP (but that is just a guess).

**Comment by sdarnell** [comment]
@mgeeforce I have rebuilt fibers.node with a thread locals limit of 40 entries, please could you try it out on WinXP? Here is the link: https://drive.google.com/file/d/0B4Vwungdd_-RX1p6cU1PZHQ3bHc/edit?usp=sharing

Download the file and rename it to `fibers.node` and replace the installed version that you have (backing up the previous one first).

If it works, I can include it in the next release of Windows meteor.

**Comment by mgeeforce** [comment]
Works!! Thanks Stephen! And actually this issue was on Win7.

**Comment by sdarnell** [comment]
@kasireddynv I have prepared an alternative version of fibers.node that outputs debug information, and this will be very helpful in pinning down the problem you are having on WinXP.
The new version is: 
https://drive.google.com/file/d/0B4Vwungdd_-ROTRiV2RIdVF5OUE/edit?usp=sharing

This version won't run fully, but will output some critical information. I get the following on Win7:

```
c:\meteor\todos>meteor
fibers: floor=10 coro_thread_key=14

fibers: floor=10 coro_thread_key=14
```

@mgeeforce I'm pleased that the increased limit seemed to fix it for you (though I have not heard of this issue on Win7 before).  It might be interesting if you could also run the debug version to see what slots you get.

Thanks, Stephen

**Comment by sdarnell** [comment]
Is seems that for some people, increasing the range of slots helps, however, for some people on Windows 8, the order of the slots is the wrong way around and prevents the current fibers from working.

There is some evidence that setting the following environment variable may help:

```
set __COMPAT_LAYER=WIN7RTM
or
set __COMPAT_LAYER=Win2000
```

It isn't clear why this setting makes a difference, but this is a relatively easy work-around if your machine hits this issue.

**Comment by talves** [comment]
@sdarnell I run win7 64 bit and have the same issue as @mgeeforce 

I ran your debug fibers.node_log and get:
fibers: floor=10 coro_thread_key=32
fibers: floor=10 coro_thread_key=32

Installed the fibers.node_with_tl_40 and all is good!
Let me know if you need anything to debug the issue.

**Comment by sdarnell** [comment]
@talves  I've just released 0.8.1.2 with a node fibers that has no limit on the thread locals range, so hopefully it should 'just work' for you now.

I've also done a bit more investigation into the other scenarios where node fibers won't work, but there is a critical assumption in node-fibers that isn't guaranteed to hold on Windows. Particularly if there DLL hooking programs installed which allocate and free thread locals. Fibers cannot handle re-use of thread locals reliably.  Maybe worth checking what Antivirus/malware and AppInit entries are installed (Autoruns can display the AppInit entries http://technet.microsoft.com/en-gb/sysinternals/bb963902.aspx ) 

For now I'm not sure there is a great deal more I can do.  So I'll close this issue.
